File: linkedlist/insert_at_last.c
#include<stdio.h>
#include<stdlib.h>
struct node{
    int data;
    struct node*next;
};
void printlinkedlist(struct node*head){
    struct node*temp=head;
    while(temp!=NULL){
     printf("%d->",temp ->data);
      temp= temp ->next;
    }
    printf("NULL");
}
void insertatlast(struct node * head,int data){
    struct node *newnode = (struct node*) malloc (sizeof(struct node));
    newnode ->data=data;
    struct node * p =head;
    
    while(p->next!=NULL){
     p=p->next;
    
    }
    newnode ->next =NULL;
    p->next= newnode;
    
}

int main(){
    struct node * head =(struct node*)malloc(sizeof(struct node));
    struct node * first =(struct node*)malloc(sizeof(struct node));
    struct node * second =(struct node*)malloc(sizeof(struct node));
    head ->data=10;
    head ->next=first;
    first ->data =20;
    first->next = second;
    second->data =30;
    second->next=NULL;
    insertatlast(head,60);
    printlinkedlist(head);
    return 0;
}

File: linked_java/insert_at_last.java
// package linked_java;

public class insert_at_last {
    class Node{
        int data;
        Node next;
        Node(int data){
            this.data=data;
            this.next=null;
        }}
        Node head = null;
        public void insert(int data){
            Node newNode = new Node(data);
            if (head==null) {
                head=newNode;
                return;
            } 
            Node temp=head;
            while (temp.next!=null) {
                temp = temp.next;
            }
            temp.next= newNode;
            // newnNode.next=null;
        }
    
    public void print(){
         Node temp=head;
            while (temp!=null) {
                System.out.print(temp.data+"->");
                temp = temp.next;
            }
        System.out.println("null");
    }
            
            

    public static void main(String[] args) {
         insert_at_last list = new insert_at_last();
        list.insert(10);
        list.insert(30);
        list.insert(40);
        list.print();
    }
}
